### PR TITLE
add expirationSeconds in CSROption to create CSR

### PR DESCRIPTION
**Pull request by youhangwang**

### CODE REVIEW

**Comment by youhangwang** [comment]
https://github.com/open-cluster-management-io/registration/issues/311

**Comment by skeeey** [comment]
@youhangwang thanks for your contributions!

Would you also help to add an option in https://github.com/open-cluster-management-io/registration/blob/main/pkg/spoke/spokeagent.go#L56

and expose the option with flags https://github.com/open-cluster-management-io/registration/blob/main/pkg/spoke/spokeagent.go#L410

**Comment by youhangwang** [comment]
yes, sure. Thanks for the review @skeeey 

please help to review the flag name.

File: pkg/spoke/spokeagent.go [hunk 1]
```diff
@@ -425,6 +428,8 @@ func (o *SpokeAgentOptions) AddFlags(fs *pflag.FlagSet) {
 		"The period to check managed cluster kube-apiserver health")
 	fs.IntVar(&o.MaxCustomClusterClaims, "max-custom-cluster-claims", o.MaxCustomClusterClaims,
 		"The max number of custom cluster claims to expose.")
+	fs.Int32Var(&o.ClientCertExpirationSeconds, "client-cert-expiration-seconds", o.ClientCertExpirationSeconds,
```
Comment (skeeey) [inline pkg/spoke/spokeagent.go:431]: I think we need validate this value in the `Validate` function https://github.com/open-cluster-management-io/registration/blob/main/pkg/spoke/spokeagent.go#L431, if this value is specified, we need require a min expiration 600s

Comment (youhangwang) [reply pkg/spoke/spokeagent.go:431]: added the validation for client-cert-expiration-seconds

**Comment by skeeey** [comment]
LGTM

@youhangwang we need a sign-off for your commits

/cc @qiujian16
please also take a look

**Comment by qiujian16** [comment]
/lgtm

Thanks for your contribution. @skeeey we need to also think how to expose these flags from operator API, same as --approve-user-list flag.

**Comment by youhangwang** [comment]
@skeeey @qiujian16 

The minimum valid value for expirationSeconds in CSR spec is 600.

But for the clientCertificateController in registration, 600 is too short:
1. clientCertificateController to check if cert is expired every 5min.
2. kubeclient resync the client cert every 5min.
3. clientCertificateController will refresh the cert if the cert has less than a random percentage range from 20% to 25% of its life remaining.

so the minimum expiration second would be (5 * 60)/(1/5) = 1500s for clientCertificateController. otherwise, the kube client will use a expired client cert to send request to the remote cluster.


Considering to add some margin. 1h(3600s) could be a good choice. Any thoughts on this?

**Comment by skeeey** [comment]
@youhangwang you are right, I forget the cert renew time,  I think we need increase the min expired time

**Comment by youhangwang** [comment]
https://github.com/open-cluster-management-io/registration/pull/320

**Pull request merged**